From b4794f7e0e61c18c6c4d3b399ad30ddd7d021576 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:35 -0400
Subject: [ethernet] mlx4: mlx4 calls skb_set_hash

Message-id: <a73fc5b1c1fb17728daed7db3207095029131499.1410425016.git.avadai@redhat.com>
Patchwork-id: 93652
O-Subject: [Patch RHEL7.1 V1 039/197] net: mlx4 calls skb_set_hash
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Tom Herbert <therbert@google.com>

Drivers should call skb_set_hash to set the hash and its type
in an skbuff.

Signed-off-by: Tom Herbert <therbert@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 6917441603bbd2a8a5cb47affbe980b2a9119b2d)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_rx.c b/drivers/net/ethernet/mellanox/mlx4/en_rx.c
index 5e7f28a..0fd1e1c 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_rx.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_rx.c
@@ -751,7 +751,9 @@ int mlx4_en_process_rx_cq(struct net_device *dev, struct mlx4_en_cq *cq, int bud
      }
 
      if (dev->features & NETIF_F_RXHASH)
-      gro_skb->rxhash = be32_to_cpu(cqe->immed_rss_invalid);
+      skb_set_hash(gro_skb,
+            be32_to_cpu(cqe->immed_rss_invalid),
+            PKT_HASH_TYPE_L3);
 
      skb_record_rx_queue(gro_skb, cq->ring);
 
@@ -796,7 +798,9 @@ int mlx4_en_process_rx_cq(struct net_device *dev, struct mlx4_en_cq *cq, int bud
    skb->encapsulation = 1;
 
   if (dev->features & NETIF_F_RXHASH)
-   skb->rxhash = be32_to_cpu(cqe->immed_rss_invalid);
+   skb_set_hash(skb,
+         be32_to_cpu(cqe->immed_rss_invalid),
+         PKT_HASH_TYPE_L3);
 
   if ((be32_to_cpu(cqe->vlan_my_qpn) &
       MLX4_CQE_VLAN_PRESENT_MASK) &&
-- 
1.7.1